File: android/app/src/main/java/vn/team/freechat/request/SendSystemMessageRequest.java
package vn.team.freechat.request;

import com.tvd12.ezyfoxserver.client.entity.EzyData;
import com.tvd12.ezyfoxserver.client.factory.EzyEntityFactory;
import com.tvd12.ezyfoxserver.client.request.EzyRequest;

import vn.team.freechat.contant.Commands;

/**
 * Created by tavandung12 on 10/5/18.
 */

public class SendSystemMessageRequest implements EzyRequest {

    private final String message;

    public SendSystemMessageRequest(String message) {
        this.message = message;
    }

    @Override
    public EzyData serialize() {
        return EzyEntityFactory.newObjectBuilder()
                .append("message", message)
                .build();
    }

    @Override
    public Object getCommand() {
        return Commands.CHAT_SYSTEM_MESSAGE;
    }
}

File: android/app/src/main/java/com/tvd12/freechat/request/SendSystemMessageRequest.kt
package com.tvd12.freechat.request

import com.tvd12.ezyfoxserver.client.entity.EzyData
import com.tvd12.ezyfoxserver.client.factory.EzyEntityFactory
import com.tvd12.ezyfoxserver.client.request.EzyRequest
import com.tvd12.freechat.constant.Commands

/**
 * Created by tavandung12 on 10/5/18.
 */

class SendSystemMessageRequest(
    private val message: String
) : EzyRequest {

    override fun serialize() : EzyData {
        return EzyEntityFactory.newObjectBuilder()
                .append("message", message)
                .build()
    }

    override fun getCommand() : String {
        return Commands.CHAT_SYSTEM_MESSAGE
    }
}
